modules/nf-core/umicollapse/main.nf

```diff
@@ -28,6 +29,8 @@ process UMICOLLAPSE {
     // which leaves 5% for stuff happening outside of java without the scheduler killing the process.
     def max_heap_size_mega = (task.memory.toMega() * 0.9).intValue()
     def max_stack_size_mega = (task.memory.toMega() * 0.05).intValue()
+    def file_type = input.getExtension().contains("fastq") ? "fastq" : "bam"
```

Comment:
What happens if somebody uses `fq` instead of `fastq` for the file names? Before, you wanted to create an argument I believe, but relying on file naming conventions to select the correct subcommand seems error-prone to me?

Reply:
Agreed. Would it better to check if it ends in `bam`?

Reply:
Possibly. But technically, I see no use case for the module to dynamically decide which subcommand needs to be run. The authors of the pipeline incorporating the module will decide. So you could probably just define a value channel as input to hardcode the run mode?

Reply:
True. This is also how some other modules like mitohifi and busco do things. User provides the mode.

Reply:
Since one probably want to allow for easier configuration without messing with complicated dynamic `ext.args`, the de facto pipeline usage will likely look like this anyway?

```
include { UMICOLLAPSE_FASTQ  } from '../../../modules/nf-core/umicollapse/main'
include { UMICOLLAPSE_BAM  } from '../../../modules/nf-core/umicollapse/main'
```